apis/refs/v1beta1/gcpserviceaccountref.go

```diff
@@ -27,7 +27,7 @@ import (
 )
 
 type IAMServiceAccountRef struct {
-	/* The GCP Service Account Email used for auth when secretType is gcpServiceAccount. Allowed value: The `email` field of an `IAMServiceAccount` resource. */
+	/* The `email` field of an `IAMServiceAccount` resource. */
```

Comment:
(Is this from some other PRs?)
I think we want to build a convention for resource reference: that the `<Kind>Ref.external`  is the GCP asset inventory-like value. So I'd vote against accepting email as the external value.
 
If it is for the backward compatibility, we can use `IAMServiceAccountRef.email` instead (warning that the `external` is deprecating in the `status`)? Not a blocker of this PR.

Reply:
So I removed the "The GCP Service Account Email used for auth when secretType is gcpServiceAccount. " prefix.  I think that's specific to one API (though I'm not sure which one).  But "The GCP Service Account Email used for auth when secretType is gcpServiceAccount. " should be on the Ref, not on external...

The problem/feature is that fixing it once here means it is fixed in all the APIs that use IAMServiceAccountRef

Reply:
Totally agree.
But still my concern is about the convention that we are using "external" to represent something other than the asset inventory format (`email`) . But as long as the IAMSARef is aligned among all its references, this is not a really big concern.